src/libraries/System.IO.Pipes/tests/NamedPipeTests/NamedPipeTest.CurrentUserOnly.Windows.cs

```diff
@@ -148,16 +185,25 @@ public NamedPipeTest_CurrentUserOnly_Windows(TestAccountImpersonator testAccount
 
                 _testAccountImpersonator.RunImpersonated(() =>
                 {
-                    using (var client = new NamedPipeClientStream(".", name, PipeDirection.InOut, clientPipeOptions))
+                    using (var client = new NamedPipeClientStream(".", name, clientDirection, clientPipeOptions))
                     {
                         Assert.Throws<UnauthorizedAccessException>(() => client.Connect());
                     }
                 });
 
-                // Server is expected to not have received any request.
-                cts.Cancel();
-                AggregateException e = Assert.Throws<AggregateException>(() => serverTask.Wait(10_000));
-                Assert.IsType<TaskCanceledException>(e.InnerException);
+                if (serverPipeOptions == PipeOptions.None && clientPipeOptions == PipeOptions.CurrentUserOnly && clientDirection == PipeDirection.In)
```

Comment:
Nit, I think it would be clearer to make this a 4th parameter to the test, so the expectation is encoded in the InlineData.

Reply:
Actually, scratch that. The expectation is essentially that all are rejected, and this is just a special case.

Reply:
I did it this way because there's only one case in this criteria. Adding another InlineData argument would look like:
false
false
true
false
false
false
false

There's nothing wrong with it, just wanted to point it out.

Reply:
Hmm, changing my mind again, because if you included success true/false, you could include
` [InlineData(PipeOptions.None, PipeOptions.None, PipeDirection.In, success:true)] ` 
which would be expected to succeed.

Reply:
if you're doing that consider also adding server PipeDirection as well. You may or may not add more permutatoins with it, but it might be easier to reason about the permutations.